src/Map/OpenGL/AtlasMapRendererSkyStage_OpenGL.cpp

```diff
@@ -61,12 +68,27 @@ bool OsmAnd::AtlasMapRendererSkyStage_OpenGL::initialize()
 
     // Compile fragment shader
     const QString fragmentShader = QLatin1String(
+        // Input data
+        "PARAM_INPUT vec2 v2f_position;                                                                                     ""\n"
+        "                                                                                                                   ""\n"
         // Parameters: common data
-        "uniform lowp vec4 param_fs_skyColor;                                                                               ""\n"
+        "uniform vec4 param_fs_skySize;                                                                                     ""\n"
+        "uniform vec4 param_fs_fogColor;                                                                                    ""\n"
         "                                                                                                                   ""\n"
         "void main()                                                                                                        ""\n"
         "{                                                                                                                  ""\n"
-        "    FRAGMENT_COLOR_OUTPUT = param_fs_skyColor;                                                                     ""\n"
+        "    vec4 color;                                                                                                    ""\n"
+        "    float position;                                                                                                ""\n"
+        "    float skyHeight = param_fs_skySize.y / (1.0f - param_fs_skySize.z);                                            ""\n"
+        "    bool sky = v2f_position.y > param_fs_skySize.z;                                                                ""\n"
+        "    position = (v2f_position.y - (sky ? param_fs_skySize.z : 0.0f)) * skyHeight;                                   ""\n"
+        "    float low = 1.0f / pow(1.1f + position, 4.0f);                                                                 ""\n"
```

Comment:
too many constants, source of formuals? matrix multiplication is easier to understand in color.rgb